src/Java.Interop.Tools.JavaTypeSystem/JavaModels/JavaMethodModel.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Java.Interop.Tools.JavaTypeSystem.Models
+{
+	public class JavaMethodModel : JavaMemberModel
+	{
+		public string Return { get; }
+		public string ReturnGeneric { get; }
+		public bool IsAbstract { get; }
+		public bool IsBridge { get; }
+		public string ReturnJni { get; }
+		public bool IsSynthetic { get; }
+		public bool IsSynchronized { get; }
+		public bool IsNative { get; }
+		public bool ReturnNotNull { get; }
+
+		public JavaTypeParameters TypeParameters { get; }
+		public JavaTypeReference? ReturnTypeModel { get; private set; }
```

Comment:
I don't know what a `Model` suffix is supposed to mean…

Reply:
This is the resolved `JavaTypeReference` for `ReturnType` string.